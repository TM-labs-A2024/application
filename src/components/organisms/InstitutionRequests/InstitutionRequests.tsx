import { HamburgerIcon } from '@chakra-ui/icons'
import { Text, IconButton, Menu, MenuButton, MenuList, MenuItem, Icon } from '@chakra-ui/react'
import RequestsList from '@components/molecules/RequestsList'
import { Doctor as DoctorType, ReactSelectOption } from '@src/types'
import { isIOS, isAndroid } from '@utils/index'
import Link from 'next/link'
import React, { useMemo } from 'react'
import { RiHealthBookFill } from 'react-icons/ri'

import Logo from '../../../../public/static/icons/logo.svg'

export default function InstitutionRequests({
  context: { doctors, specialtiesOptions }
}: {
  context: {
    doctors: DoctorType[]
    specialtiesOptions: ReactSelectOption[]
  }
}) {
  // --- Data and handlers -----------------------------------------------------
  const formatedRequests = useMemo(
    () =>
      doctors.map(({ id, govId, firstname, lastname, specialties, requestId }) => ({
        href: `/institucion/medico/${id}/${requestId}`,
        title: `${firstname} ${lastname}`,
        description: `C.I: ${govId},${specialties.map((specialty) => ' ' + specialtiesOptions.find((el) => el?.value === specialty.id)?.label)}.`
      })),
    [doctors, specialtiesOptions]
  )
  // --- END: Data and handlers ------------------------------------------------

  return (
    <div
      className={`mx-auto block h-screen w-screen overflow-hidden px-8 ${isIOS() ? 'pb-24 pt-20' : 'pb-24 pt-10'} ${isAndroid() ?? 'pb-64 pt-8'}`}
      data-testid="institution-requests"
    >
      <div className={`mb-8 flex flex-row justify-between`}>
        <Logo />
        <Menu>
          <MenuButton
            as={IconButton}
            aria-label="Options"
            size="xl"
            fontSize="30px"
            variant="link"
            icon={<HamburgerIcon />}
            onClick={() => null}
          />
          <MenuList>
            <MenuItem>
              <Link href="/institucion/medicos">Médicos con acceso</Link>
            </MenuItem>
            <MenuItem>
              <Link href="/perfil/institucion">Perfil de la institución</Link>
            </MenuItem>
          </MenuList>
        </Menu>
      </div>
      {formatedRequests.length > 0 && (
        <div className="h-full lg:px-80">
          <RequestsList requests={formatedRequests} label="Solicitudes pendientes" />
        </div>
      )}
      {formatedRequests.length === 0 && (
        <div
          className="flex h-3/4 w-full flex-col items-center justify-center"
          data-testid="pending-doctors-empty-state"
        >
          <Icon fontSize="xxx-large" as={RiHealthBookFill} />
          <Text textAlign="center" mt={4}>
            No hay solicitudes aún.
          </Text>
        </div>
      )}
    </div>
  )
}
