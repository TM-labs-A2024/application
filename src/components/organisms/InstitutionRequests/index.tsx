import InstitutionRequests from './InstitutionRequests'

export default InstitutionRequests
