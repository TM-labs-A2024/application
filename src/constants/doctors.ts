export const doctors = [
  {
    id: 1000000,
    specialities: [1],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 2000000,
    specialities: [2, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 3000000,
    specialities: [3, 5, 7],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 4000000,
    specialities: [4, 1],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 5000000,
    specialities: [5, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 6000000,
    specialities: [6, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 7000000,
    specialities: [7, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 8000000,
    specialities: [8, 9, 5, 1, 3, 4],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 9000000,
    specialities: [9, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 10000000,
    specialities: [10, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 11000000,
    specialities: [11, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 12000000,
    specialities: [12, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 13000000,
    specialities: [13, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 14000000,
    specialities: [14, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 15000000,
    specialities: [15, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 16000000,
    specialities: [16, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 17000000,
    specialities: [17, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 18000000,
    specialities: [18, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 19000000,
    specialities: [19, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 20000000,
    specialities: [20, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 21000000,
    specialities: [21, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 22000000,
    specialities: [22, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522',
    pending: true
  },
  {
    id: 23000000,
    specialities: [23, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 24000000,
    specialities: [24, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 25000000,
    specialities: [25, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 26000000,
    specialities: [26, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 27000000,
    specialities: [27, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 28000000,
    specialities: [28, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 29000000,
    specialities: [29, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 30000000,
    specialities: [30, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 31000000,
    specialities: [31, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 32000000,
    specialities: [32, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 33000000,
    specialities: [33, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 34000000,
    specialities: [34, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 35000000,
    specialities: [35, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 36000000,
    specialities: [36, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 37000000,
    specialities: [37, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 38000000,
    specialities: [38, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 39000000,
    specialities: [39, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 40000000,
    specialities: [40, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 41000000,
    specialities: [41, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 42000000,
    specialities: [42, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 43000000,
    specialities: [43, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 44000000,
    specialities: [44, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 45000000,
    specialities: [45, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 46000000,
    specialities: [46, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 47000000,
    specialities: [47, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  },
  {
    id: 48000000,
    specialities: [48, 9, 5],
    firstname: 'José',
    lastname: 'Pérez',
    birthdate: 'Wed Sep 17 1997 00:00:00 GMT-0400 (hora de Venezuela)',
    email: 'joseperez@gmail.com ',
    phone: '4127775522'
  }
]
