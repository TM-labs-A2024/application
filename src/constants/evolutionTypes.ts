export const evolutionTypes = [
  {
    id: '1',
    name: 'Evolución'
  },
  {
    id: '2',
    name: 'Hospitalización'
  },
  {
    id: '3',
    name: 'Alta'
  },
  {
    id: '4',
    name: 'Primera consulta'
  }
]
