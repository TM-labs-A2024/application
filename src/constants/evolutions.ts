export const evolution = {
  specialty: {
    description: '',
    id: '',
    name: ''
  },
  author: '',
  title: '',
  description: '',
  'content-format': '',
  created_at: '',
  payload: {
    type: {
      value: '',
      label: ''
    },
    reason: '',
    description: '',
    history: '',
    examination: '',
    summary: '',
    diagnostic: '',
    comments: ''
  }
}
