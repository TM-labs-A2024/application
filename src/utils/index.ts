import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
