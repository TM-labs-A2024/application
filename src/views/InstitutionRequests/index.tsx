import InstitutionRequests from '@components/organisms/InstitutionRequests'
import React from 'react'

export default function InstitutionRequestsView() {
  return <InstitutionRequests />
}
