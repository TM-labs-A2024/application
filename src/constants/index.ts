import { doctors } from './doctors'
import { evolutionTypes } from './evolutionTypes'
import { institution } from './institution'
import { institutions } from './institutions'
import {
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
} from './notifications'
import { nurse } from './nurses'
import { patient, sexOptions, statusOptions } from './patient'
import { patients } from './patients'
import { specialties, specialities } from './specialties'
import { specialtyData } from './specialtyData'

export {
  specialties,
  specialities,
  doctors,
  nurse,
  patient,
  sexOptions,
  statusOptions,
  specialtyData,
  evolutionTypes,
  patients,
  institution,
  institutions,
  ACCESS_REMOVED,
  FILTERS_APPLIED,
  ACCESS_REQUESTED,
  ATTACHMENT_DELETED,
  IMAGE_DELETED,
  ACCESS_DENIED,
  ACCESS_GRANTED,
  ACCESS_REMOVAL,
  GENERIC_NOTIFICATION,
  GENERIC_ERROR
}
