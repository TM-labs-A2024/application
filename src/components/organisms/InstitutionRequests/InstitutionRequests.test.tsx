import { doctors } from '@src/constants'
import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import InstitutionRequests from './InstitutionRequests'

const pendingDoctors = doctors.filter((doctor) => doctor.pending)

// El renderizado del componente concuerda con la captura instantánea del componente

describe('Organisms > InstitutionRequests test', () => {
  test('The component shows the list of pending doctors if it has at least one doctor', () => {
    render(<InstitutionRequests doctors={pendingDoctors} />)

    expect(screen.getByText('Solicitudes pendientes')).toBeInTheDocument()
  })

  test('The component shows the empty state if the list of pending doctors is empty', () => {
    render(<InstitutionRequests doctors={[]} />)

    expect(screen.getByTestId('pending-doctors-empty-state')).toBeInTheDocument()
  })

  test("The component allows to go to the institution's profile", () => {
    render(<InstitutionRequests doctors={pendingDoctors} />)

    expect(screen.getByText('Perfil de la institución')).toHaveAttribute(
      'href',
      '/perfil/institucion'
    )
  })

  test("The component allows to go to the institution's pending requests page", () => {
    render(<InstitutionRequests doctors={pendingDoctors} />)

    expect(screen.getByText('Médicos con acceso')).toHaveAttribute('href', '/institucion/medicos')
  })

  test('Matches the snapshot', () => {
    render(<InstitutionRequests doctors={pendingDoctors} />)

    expect(screen.getByTestId('institution-requests')).toMatchSnapshot()
  })
})
