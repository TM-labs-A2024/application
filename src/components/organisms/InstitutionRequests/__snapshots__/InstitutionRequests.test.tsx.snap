// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms > InstitutionRequests test Matches the snapshot 1`] = `
<div
  class="mx-auto block h-screen w-screen overflow-hidden px-8 pb-24 pt-10 false"
  data-testid="institution-requests"
>
  <div
    class="mb-8 flex flex-row justify-between"
  >
    <div />
    <button
      aria-controls="menu-list-:rh:"
      aria-expanded="false"
      aria-haspopup="menu"
      aria-label="Options"
      class="chakra-button chakra-menu__menu-button css-wpzvz6"
      id="menu-button-:rh:"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="css-r6z5ec"
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-1kfu8nn"
        id="menu-list-:rh:"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
        tabindex="-1"
      >
        <button
          class="chakra-menu__menuitem css-18esm8n"
          data-index="0"
          id="menu-list-:rh:-menuitem-:ri:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <a
            href="/institucion/medicos"
          >
            Médicos con acceso
          </a>
        </button>
        <button
          class="chakra-menu__menuitem css-18esm8n"
          data-index="1"
          id="menu-list-:rh:-menuitem-:rj:"
          role="menuitem"
          tabindex="-1"
          type="button"
        >
          <a
            href="/perfil/institucion"
          >
            Perfil de la institución
          </a>
        </button>
      </div>
    </div>
  </div>
  <div
    class="h-full lg:px-80"
  >
    <div
      class="chakra-stack h-full css-8g8ihq"
      data-testid="requests-list"
    >
      <h2
        class="chakra-heading mb-4 css-9f6g39"
      >
        Solicitudes pendientes
      </h2>
      <div
        class="overflow-auto"
      >
        <ul
          class="css-155za0w"
          role="list"
        >
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/1000000/1000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 1000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/2000000/2000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 2000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/3000000/3000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 3000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/4000000/4000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 4000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/5000000/5000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 5000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/6000000/6000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 6000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/7000000/7000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 7000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/16000000/16000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 16000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/17000000/17000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 17000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/18000000/18000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 18000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/19000000/19000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 19000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/20000000/20000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 20000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/21000000/21000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 21000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
          <div
            class="border-b border-black py-4 last:border-transparent"
          >
            <a
              class="chakra-link w-full css-spn4bz"
              data-testid="request-card"
              href="/institucion/medico/22000000/22000000"
            >
              <div
                class="flex flex-row items-center justify-between"
              >
                <div
                  class="flex flex-col items-start gap-1"
                >
                  <b
                    class="chakra-text css-qqfgvy"
                  >
                    José Pérez
                  </b>
                  <p
                    class="chakra-text css-0"
                  >
                    C.I: 22000000, undefined, undefined.
                  </p>
                </div>
                <div />
              </div>
            </a>
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`
